independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/ListExtensionsCommandHandler.java

```diff
@@ -49,15 +51,19 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
         } catch (IOException e) {
             throw new QuarkusCommandException("Failed to determine the list of installed extensions", e);
         }
-
-        Stream<Extension> platformExtensionsStream = invocation.getPlatformDescriptor().getExtensions().stream();
-        platformExtensionsStream = platformExtensionsStream.filter(this::filterUnlisted);
-        if (search != null && !"*".equalsIgnoreCase(search)) {
-            final Pattern searchPattern = Pattern.compile(".*" + search + ".*", Pattern.CASE_INSENSITIVE);
-            platformExtensionsStream = platformExtensionsStream.filter(e -> filterBySearch(searchPattern, e));
+        final Collection<Extension> platformExtensions;
+        if (extensionRegistry != null) {
+            String quarkusVersion = invocation.getPlatformDescriptor().getQuarkusVersion();
+            platformExtensions = extensionRegistry.list(quarkusVersion, search);
+        } else {
+            Stream<Extension> platformExtensionsStream = invocation.getPlatformDescriptor().getExtensions().stream();
+            platformExtensionsStream = platformExtensionsStream.filter(this::filterUnlisted);
+            if (search != null && !"*".equalsIgnoreCase(search)) {
+                final Pattern searchPattern = Pattern.compile(".*" + search + ".*", Pattern.CASE_INSENSITIVE);
+                platformExtensionsStream = platformExtensionsStream.filter(e -> filterBySearch(searchPattern, e));
+            }
+            platformExtensions = platformExtensionsStream.collect(Collectors.toList());
         }
-        List<Extension> platformExtensions = platformExtensionsStream.collect(Collectors.toList());
-
         if (platformExtensions.isEmpty()) {
             invocation.log().info("No extension found with this pattern");
```

Comment:
This message could be more informative. Like:

"No extension found with pattern `kubernetes` when searching https://github.com/quarkusio/quarkus-extension-catalog/releases/download/v23/registry.json"

or similar.